jeneamocan/script.rb

```diff
@@ -0,0 +1,204 @@
+      puts "Authentication successful"
+    else
+      puts "Invalid username or password, try again"
+      authentication
+    end
+  end
+
+  def authentication_check
+    if browser.text_field(name: "login").present?
+      puts "Authentication required"
+      authentication
+    else
+    end
+  end
+
+  def info
+    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.NEW_CARDS_ACCOUNTS")
+    name = browser.div(class: "main-info").a(class: "name").text
+    balance = browser.div(class: "primary-balance").span(class: "amount").text
+    currency = browser.div(class: "primary-balance").span(class: %w"amount currency").text
+    nature = browser.div(class: %w"section-title h-small").text.downcase.capitalize
+    puts ["**Account**", "- #{name}", "- #{currency}", "- #{balance}", "- #{nature}"]
+  end
+
+  def parse
+    Nokogiri::HTML.parse(browser.html)
+  end
+
+  def store_result
+    if File.exist?('data/accounts.json') && File.exist?('data/transactions.json')
+      accounts = File.read('data/accounts.json')
+      transactions = File.read('data/transactions.json')
+      accounts_json = JSON.parse(accounts)
+      transactions_json = JSON.parse(transactions)
+      result = accounts_json
+      result["accounts"][0]["transactions"] = transactions_json["transactions"]
+      result
+      Dir.mkdir(DATA_DIR) unless File.exists?(DATA_DIR)
+      file_name = "#{DATA_DIR}/#{File.basename('result')}.json"
+      File.open(file_name, 'w'){|file| file.write(JSON.pretty_generate(result))}
+      puts "Accounts with transactions saved to #{file_name}"
+    else
+      puts "Information is absent"
+    end
+  end
+end
+
+class Accounts < VB_WebBanking
+  attr_reader :account
+
+  def initialize
+    run
+    account_info
+    store
+  end
+
+  def account_info
+    @account = account_check
+  end
+
+  def account_check
+    authentication_check
+    puts "Fetching account informatrion"
+    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.NEW_CARDS_ACCOUNTS")
+    page = parse
+    hash = {"accounts" => []}
+    i = page.css('div[class="contracts-section"]').map do |i|
+      unless i.css('div[class="section-title no-data-error"]').any?
+        name = i.css('div[class="main-info"]').css('a[class="name"]').text
+        balance = i.css('div[class="primary-balance"]').css('span[class="amount"]').text
+        currency =  i.css('div[class="primary-balance"]').children.last.text
+        nature = i.css('div[class="section-title h-small"]').text.downcase.capitalize
+        res = {name: name, currency: currency, balance: balance, nature: nature}
+        hash["accounts"] << res
+      else
+      end
+    end
+    hash
+  end
+
+  def store
+    Dir.mkdir(DATA_DIR) unless File.exists?(DATA_DIR)
+    file_name = "#{DATA_DIR}/#{File.basename('accounts')}.json"
+    File.open(file_name, 'w'){|file| file.write(JSON.pretty_generate(@account))}
+    puts "Accounts saved to #{file_name}"
+  end
+end
+
+class Transactions < VB_WebBanking
```

Comment:
same as with Account class. When you run your script leave script class call in the last lines like 
```ruby
parser = VB_WebBanking.new(from_date, to_date)
parser.run
puts your_file_with_output
```